python-package/lightgbm/callback.py

```diff
@@ -164,6 +167,7 @@ def early_stopping(stopping_rounds, verbose=True):
     best_iter = []
     best_msg = []
     cmp_op = []
+    score_list = []
```

Comment:
I think you should use dict here, same format as record_evaluation: {dataset name: {eval metric name: score}}, otherwise user can't know the order of those metrics